tvm/src/pass/generate_reuse_buffer.cc

```diff
@@ -55,6 +55,124 @@ class ModulusRemover final : public IRMutator {
   std::map<const Variable*, Expr>& range_;
 };
 
+// Remove cast in binary expressions
+// Example: (cast(x) + cast(y)) -> (x + y)
+// Usage: CastRemover castRemover;
+//        Expr expr = castRemover.Mutate(expr);
+class CastRemover final : public IRMutator {
```

Comment:
I would suggest moving this CastRemover class to `ir_utils.cc`, so that it can also be used in some other passes.